JT/13_Applied_OOP/Applied_OOP.adoc

```diff
@@ -0,0 +1,189 @@
+== Applied OOP
+
+Author Name: Igor Chele 28.03.2024
+
+=== - Encapsulation
+
+Encapsulation is (one of) the fundamental principle of object-oriented programming (OOP).
+
+It refers to the bundling of data (attributes or properties) and methods (functions or procedures) that operate on the data into a single unit called a class.
+Encapsulation hides the internal state of an object from the outside world and only exposes the necessary functionalities through well-defined interfaces.
+This helps in ensuring data integrity, promoting code modularity, and reducing system complexity.
+
+To achieve this, you must:
+
+* declare class variables/attributes as *private*
+* provide public *get* and *set* methods to access and update the value of a private variable
+
+[source,java]
+----
+public class Person {
+    private String name;
+    private int age;
+    private String gender;
+
+
+    public String getName() {
+        return name;
+    }
+
+    public void setName(String name) {
+        this.name = name;
+    }
+
+    public int getAge() {
+        return age;
+    }
+
+    public void setAge(int age) {
+        this.age = age;
+    }
+
+    public String getGender() {
+        return gender;
+    }
+
+    public void setGender(String gender) {
+        this.gender = gender;
+    }
+
+----
+
+****
+See a full example attached in the link:Examples/Encapsulation/[Encapsulation] package.
+****
+
+Also , strongly recomended to watch these bedtime stories :)
+https://www.youtube.com/watch?v=Mb1TJprZb1c[Encapsulation] and https://www.youtube.com/watch?v=WZEPT77pB0g[Encapsulation aplication], useful not only for encapsulation.
```

Comment:
please embed video in the adoc. as an example: `video::Mb1TJprZb1c[youtube]`
and the video will be displayed in that way:
![image](https://github.com/bodacd/automation-study-materials/assets/132050848/c722e3b5-5c79-4ac4-b478-e2772e774c69)
also, you can play with the displaying of media player

Reply:
Tried, but in this format it gives an error on click, while as link it opens a browser tab with YouTube.

Reply:
I added both methods,